daemon/src/main/java/org/sonarlint/daemon/services/StandaloneSonarLintImpl.java

```diff
@@ -19,37 +19,159 @@
+import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneAnalysisConfiguration;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration.Builder;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneSonarLintEngine;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.AnalysisReq;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.InputFile;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue.Severity;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.LogEvent;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.RuleDetails;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.RuleKey;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.StandaloneConfiguration;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Void;
-import org.sonarsource.sonarlint.daemon.proto.StandaloneSonarLintGrpc.StandaloneSonarLint;
-
-import java.net.URL;
-import java.nio.charset.Charset;
-import java.nio.file.Paths;
-import java.util.Arrays;
-import java.util.LinkedList;
-import java.util.List;
+import org.sonarsource.sonarlint.daemon.proto.StandaloneSonarLintGrpc;
 
-public class StandaloneSonarLintImpl extends AbstractSonarLint implements StandaloneSonarLint {
+public class StandaloneSonarLintImpl extends StandaloneSonarLintGrpc.StandaloneSonarLintImplBase {
   private static final Logger LOGGER = LoggerFactory.getLogger(StandaloneSonarLintImpl.class);
   private StandaloneSonarLintEngine engine;
   private ProxyLogOutput logOutput = new ProxyLogOutput();
 
+  static class DefaultClientInputFile implements ClientInputFile {
+    private final Path path;
+    private final boolean isTest;
+    private final Charset charset;
+    private final String userObject;
+
+    protected DefaultClientInputFile(Path path, boolean isTest, Charset charset, String userObject) {
+      this.path = path;
+      this.isTest = isTest;
+      this.charset = charset;
+      this.userObject = userObject;
+    }
+
+    @Override
+    public Path getPath() {
+      return path;
+    }
+
+    @Override
+    public boolean isTest() {
+      return isTest;
+    }
+
+    @Override
+    public Charset getCharset() {
+      return charset;
+    }
+
+    @Override
+    public <G> G getClientObject() {
+      return (G) userObject;
+    }
+  }
+
+  static class ProxyIssueListener implements IssueListener {
+    private final StreamObserver<Issue> observer;
+
+    protected ProxyIssueListener(StreamObserver<Issue> observer) {
+      this.observer = observer;
+    }
+
+    @Override
+    public void handle(org.sonarsource.sonarlint.core.client.api.common.analysis.Issue issue) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "handle" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)